### PR TITLE
Fix the CBMC proof of ARPProcessPacket

### PR DESCRIPTION
**Pull request by AniruddhaKanhere**
Description
-----------

Fix the ARPProcessPacket CBMC proof.

This change was required since the functioning has changed as compared to the IPv4 code base. There is a new concept of Interfaces and Endpoints which needs to be addressed in the proofs dealing with them.
This should technically be considered as the proofs being *added* rather than the proofs being *changed* since this is a new library (although based on IPv4 code base).

Test Steps
-----------

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/cbmc/proofs/ARP/ARPProcessPacket/ARPProcessPacket_harness.c [hunk 1]
```diff
@@ -6,10 +6,63 @@
...
+const uint8_t ucDNSServerAddress2[ 4 ];
+const uint8_t ucMACAddress[ 6 ];
 
 void harness()
 {
+    NetworkBufferDescriptor_t xLocalNetworkBufferDescriptor;
     ARPPacket_t xARPFrame;
+    NetworkEndPoint_t xLocalEndPoint;
+
+    /* The Ethernet buffer must contain the ARP packet. */
```
Comment (markrtuttle) [inline test/cbmc/proofs/ARP/ARPProcessPacket/ARPProcessPacket_harness.c:28]: Nice comment, I like that.

File: test/cbmc/proofs/ARP/ARPProcessPacket/ARPProcessPacket_harness.c [hunk 2]
```diff
@@ -6,10 +6,63 @@
...
+    xLocalNetworkBufferDescriptor.pucEthernetBuffer = &xARPFrame;
+
+    /* Non-deterministically add an end-point to the descriptor. */
+    if( nondet_bool() )
+    {
+        xLocalNetworkBufferDescriptor.pxEndPoint = NULL;
```
Comment (markrtuttle) [inline test/cbmc/proofs/ARP/ARPProcessPacket/ARPProcessPacket_harness.c:34]: Can you use conditional assignment x = nondet_bool() ? NULL : expression

?

**Review by markrtuttle — CHANGES_REQUESTED** [review]

File: test/cbmc/proofs/ARP/ARPProcessPacket/ARPProcessPacket_harness.c (hunk 2 above)
Comment (AniruddhaKanhere) [reply test/cbmc/proofs/ARP/ARPProcessPacket/ARPProcessPacket_harness.c:34]: Sure! that works for me :)

File: test/cbmc/proofs/ARP/ARPProcessPacket/ARPProcessPacket_harness.c (hunk 1 above)
Comment (AniruddhaKanhere) [reply test/cbmc/proofs/ARP/ARPProcessPacket/ARPProcessPacket_harness.c:28]: Thanks!

**Review by markrtuttle — APPROVED** [review]

**Review by dan4thewin — APPROVED** [review]

**Pull request merged**